lambdaclass/mirra_backend
# Implement defense

[Issue](https://github.com/lambdaclass/champions_of_mirra/issues/326)

Adds defense to DealDamage Executions according to the [game design](https://www.notion.so/lambdaclass/Mechanics-fda91b86b0cf450d92ff08fd08ea8813).

A necessary change has been added where we abstract the TestUtils, so that we could use them for our new Defense test (and future ones as well)

# To Test

Run a battle with Muflus like detailed on `battle.md`. Check the history output for EFFECT_HIT of basic skills. They should be dealing less damage now:
```
          action_type: {:skill_action,
           %{
             caster_id: "e119b3b4-e189-42f0-9967-373bf2639881",
             skill_id: "037ec7ec-466c-4f47-998e-5bbae261cae7",
             skill_action_type: :EFFECT_HIT,
             stats_affected: [%{stat: :HEALTH, amount: -28}],
             target_ids: ["e66eec75-17ca-4c04-a0ac-2b3865c98580"]
           }}
        }
```
They were dealing 50 damage before.
